### PR TITLE
M680X: Target ready for pull request

### PR DESCRIPTION
**Pull request by aladur**
Meanwhile I heavily extended this new target to support these microprocessors/controllers:
        CS_MODE_M680X_6301 = 1 << 1, // M680X Hitachi 6301,6303 mode
        CS_MODE_M680X_6309 = 1 << 2, // M680X Hitachi 6309 mode
        CS_MODE_M680X_6800 = 1 << 3, // M680X Motorola 6800,6802 mode
        CS_MODE_M680X_6801 = 1 << 4, // M680X Motorola 6801,6803 mode
        CS_MODE_M680X_6805 = 1 << 5, // M680X Motorola/Freescale 6805 mode
        CS_MODE_M680X_6808 = 1 << 6, // M680X Motorola/Freescale/NXP 68HC08 mode
        CS_MODE_M680X_6809 = 1 << 7, // M680X Motorola 6809 mode
        CS_MODE_M680X_6811 = 1 << 8, // M680X Motorola/Freescale/NXP 68HC11 mode
        CS_MODE_M680X_CPU12 = 1 << 9, // M680X Motorola/Freescale/NXP CPU12
                                        // used on M68HC12/HCS12
        CS_MODE_M680X_HCS08 = 1 << 10, // M680X Freescale/NXP HCS08 mode
Try test_m680x to check how it works.
There are bindings for python, java and ocaml available. 
See also my test project: https://github.com/aladur/capstone_m680x_test
According to my google results the microcontrollers 68HC08, 68HC11 and CPU12 are still under production by NXP.

### CODE REVIEW

**Comment by aquynh** [comment]
this is fantastic! i will get back with comments soon, thanks.

File: bindings/python/capstone/m680x.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Capstone Python bindings, by Nguyen Anh Quynnh <aquynh@gmail.com>
```
Comment (aquynh) [inline bindings/python/capstone/m680x.py]: please put your name here.

File: bindings/python/test_m680x.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+
+# Capstone Python bindings, by Nguyen Anh Quynnh <aquynh@gmail.com>
```
Comment (aquynh) [inline bindings/python/test_m680x.py]: please put your name on the new files you created.

File: cs_priv.h [hunk 3]
```diff
@@ -76,7 +76,7 @@ struct cs_struct {
 	struct insn_mnem *mnem_list;	// linked list of customized instruction mnemonic
 };
 
-#define MAX_ARCH 10
+#define MAX_ARCH 11
```
Comment (aquynh) [inline cs_priv.h]: we can change this line to:

```
#define MAX_ARCH CS_ARCH_MAX
```

**Comment by aladur** [comment]
Commited changes on your comments.

**Comment by aquynh** [comment]
awesome work, merged, thanks!

**Comment by aquynh** [comment]
hope to see @radare pick up these new archs for the next r2 version.

**Comment by aquynh** [comment]
@aladur, to be consistent with other architectures, can you put a space after each comma in operand string?

currently the output is like this:

```
PSHS            CC,B,X,U
```

but it should be like:

```
PSHS            CC, B, X, U
```

thanks.

**Pull request merged**